### PR TITLE
feat: Publishing offchain future assets

### PR DESCRIPTION
**Pull request by akhercha**
## Added
- new linters: `ruff` + `mypy` for types

## Changed
- publish_data to offchain future entries

## Removed
- old linters installations & rules 

________

Currently, `mypy` ignore every source files (see pyproject.toml), an issue will be created to slowly fix them all (there is ~700 atm).
See #116

### CODE REVIEW

File: pragma/publisher/perp_fetchers/binance.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+                    expiry_timestamp = int(0)
+            result_arr.append(
+                FutureEntry(
+                    pair_id=pair_id,
+                    price=price_int,
+                    volume=int(volume),
+                    timestamp=int(timestamp / 1000),
+                    source=self.SOURCE,
+                    publisher=self.publisher,
+                    expiry_timestamp=expiry_timestamp * 1000,
```
Comment (EvolveArt) [inline pragma/publisher/perp_fetchers/binance.py:120]: its a perp fetcher and still has an expiry?

File: pragma/publisher/perp_fetchers/binance.py [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+                    autoscale_volume=True,
```
Comment (EvolveArt) [inline pragma/publisher/perp_fetchers/binance.py:121]: double check if the volume needs to be autoscaled

File: pragma/core/mixins/offchain.py [hunk 3]
```diff
@@ -128,10 +136,12 @@ async def publish_data(
 
         body = {
             "signature": [str(s) for s in sig],
-            "entries": SpotEntry.offchain_serialize_entries(entries),
+            "entries": EntryClass.offchain_serialize_entries(entries),
         }
 
         url = self.api_url + "/v1/data/publish"
+        if isinstance(entries[0], FutureEntry):
```
Comment (EvolveArt) [inline pragma/core/mixins/offchain.py:143]: create a function `get_api_url(data_type)`

**Comment by EvolveArt** [comment]
Please also add some tests for the new perp fetcher.

File: pragma/publisher/perp_fetchers/binance.py (hunk 1 above)
Comment (akhercha) [reply pragma/publisher/perp_fetchers/binance.py]: ooops, copy/paste error, not it's always zero

**Comment by akhercha** [comment]
> Please also add some tests for the new perp fetcher.

Removed the PerpFetcher for now as discussed in DM - since I can't test properly cause tests don't work locally

**Review by EvolveArt — APPROVED** [review]

**Pull request merged**